### ISSUE TITLE
[bug] You can't cancel option setting

### ISSUE THREAD

**Issue by lynxlynxlynx** [gemrb/gemrb#535]
The way we implemented GUIOPT option saving, it is done immediately after confirmation. So subwindows are treated separately. But also since the game variables are changed directly, it's currently impossible to cancel setting the options. The buttons don't do anything.

Maybe we need guiscript store/load gamevar functions, which we'd use like db transactions?

**Comment by bradallred** [comment]
speaking of db transactions... isn't sqlite what the EEs use anyway? If we will have a need for sql to support those, it might make sense to just transition to that for our settings as well.

**Comment by MarcelHB** [comment]
Sounds like a bit of overkill. Not sure if and how EE uses sqlite, but isn't that way more boilerplate over reading the settings into a copy, and later write it back?

**Comment by lynxlynxlynx** [comment]
It's some odd sql flavour that they use for some settings and journal entries ... in some versions. We'll likely be able to avoid it all.

**Comment by bradallred** [comment]
yeah, I'm not sure what they use it for, I just know it is there :man_shrugging: 

If we have no need for it then we have no need for it. 

But to address the boilerplate question, that totally depends on which bindings you use. There are several available that make reading and writing a value just a single line of code, but also comes with the benefit of supporting transactions and data validation though the db schema which is something we have to do manually currently.

**Comment by MarcelHB** [comment]
Good to know, I've never looked at a single C++ ORM lib.

**Comment by bradallred** [comment]
Now that you mention it I don't think I have either ha ha
(unless we are counting proprietary employer ones)

**Comment by lynxlynxlynx** [comment]
@bradallred your last commit mentioned "*this will be broken until the global variable hack in Window::SetDictVariable is removed*". We never had such a function, maybe that's something from one of the unfinished PRs. Overall, can this be closed now or is anything remaining to be done?

**Comment by bradallred** [comment]
that commit is part of #1788 and not one of the pieces I recently cherry-picked into #1876 either.
